### DISCUSSION TITLE
Scrollbar in markdown element

### DISCUSSION THREAD

**Discussion by dofdary** [zauberzeug/nicegui#1185] [Q&A] [answered]
### Question

Hello,
I have several markdown elements on my page showing some json data.
The json is big, therefore I would like to have both vertical and horizontal scrolling for each markdown element.

The markdown elements are on a grid.
Code:

```python
import json

from nicegui import ui


def index():
    obj = {"a": "bc", "b": ["a", {"mmh": "bc "*20}]}
    formatted = json.dumps(obj, indent=2)

    with ui.card():
        with ui.grid(columns=2):
             for _ in range(6):
                with ui.element("div").classes("p-2 rounded-borders shadow-2").style("background-color: darkgrey"):
                    ui.markdown(f"```json\n{formatted}\n```")


index()
ui.run(
    title="Scroll Problem",
    favicon="📜",
    port=8080,
    dark=None,
    binding_refresh_interval=0.5,
    show=False,
    reload=False
)
```

Result:

<img width="1719" alt="renderNoScroll" src="https://github.com/zauberzeug/nicegui/assets/129761296/21da550b-df47-49ad-a9e5-e89a8f06abdd">

I then added a `scroll_area` element to contain the markdown:
```python
                    with ui.scroll_area():
                        ui.markdown(f"```json\n{formatted}\n```")
```

Result:

<img width="1724" alt="renderScroll1" src="https://github.com/zauberzeug/nicegui/assets/129761296/771557be-6267-4f6e-b846-864c3328ed84">

This is not satisfying because now the markdown elements look like noodles

So I followed the example in the doc and added a width class to my div (w-32):

```python
                with ui.element("div").classes("p-2 rounded-borders shadow-2 w-32").style("background-color: darkgrey"):
                    with ui.scroll_area():
                        ui.markdown(f"```json\n{formatted}\n```")
```

Result:

<img width="1713" alt="renderScroll2" src="https://github.com/zauberzeug/nicegui/assets/129761296/4f2a9949-4e25-4348-a4a6-119fd0fafb8d">

This is better but still not what I wanted, because now the width of the markdown elements is fixed.
I would like each markdown to take half the space offered by the parent element (like in my first picture, before the `scroll_area`).
Can someone help me to achieve this?

Thanks in advance.

**Comment by falkoschindler** [comment, marked as answer]
Hi @dofdary!

Your noodles made my day! 🍜😄

The trick is to let the `ui.card` and the `ui.grid` fill the horizontal space:
```py
with ui.card().classes('w-full'):
    with ui.grid(columns=2).classes('w-full'):
        for _ in range(6):
            with ui.element().classes("p-2 rounded-borders shadow-2 bg-[darkgrey]"):
                with ui.scroll_area():
                    ui.markdown(f"```json\n{formatted}\n```")
```

Apart from that the `ui.element` and `ui.scroll_area` can be merged into one element:
```py
with ui.scroll_area().classes("p-2 rounded-borders shadow-2 bg-[darkgrey]"):
```

Oh and you might not need the scroll element at all. You can simply set `overflow-auto` for the Markdown element and it will scroll if needed:
```py
ui.markdown(f"```json\n{formatted}\n```").classes("... overflow-auto")
```

**Reply by dofdary** [reply]
Working flawlessly thank you!
I chose the solution without scroll_area. What does the triple dots mean in the classes of the markdown?

**Reply by falkoschindler** [reply]
Great!
Oh, I just didn't want to repeat all classes from above. You can replace "..." with whatever classes you like. 🙂
